apps/daimo-mobile/src/action/useLandlineDeposit.ts

```diff
@@ -0,0 +1,49 @@
+import { daimoChainFromId } from "@daimo/contract";
+import * as Haptics from "expo-haptics";
+import { useCallback } from "react";
+
+import { ActHandle, useActStatus } from "../action/actStatus";
+import { getRpcFunc } from "../logic/trpc";
+import { Account } from "../storage/account";
+
+interface UseLandlineDepositArgs {
+  account: Account;
+  recipient: { landlineAccountUuid: string };
+  dollarsStr: string;
+  memo?: string;
+}
+
+export function useLandlineDeposit({
+  account,
+  recipient,
+  dollarsStr,
+  memo,
+}: UseLandlineDepositArgs): ActHandle & { exec: () => Promise<void> } {
+  const [as, setAS] = useActStatus("useLandlineDeposit");
+
+  const exec = useCallback(async () => {
+    setAS("loading", "Creating deposit");
+    try {
+      // TODO: authenticate call to the deposit RPC function
```

Comment:
this is the part where we need session keys

Reply:
alternative: we can use an ERC-1271 signed message, just like we do in the Link Farcaster flow.

will require FaceID, but that's a feature IMO: feels correct to authenticate equally for (deposit from bank) vs (withdraw to bank)

Reply:
should we ship a version with deposits disabled and then create a follow-up PR with authentication for deposits?

Reply:
nvm, added offchain action for authentication